FocusOnTests/ProgressViewModelTests.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ProgressViewModelTests.swift
+//  FocusOnTests
+//
+//  Created by Alexandra Ivanova on 05/08/2023.
+//
+
+import XCTest
+import SwiftUICharts
+@testable import FocusOn
+
+class ProgressViewModelTests: XCTestCase {
+    
+    var viewModel: ProgressViewModel!
+    
+    override func setUpWithError() throws {
+        viewModel = ProgressViewModel(dataService: MockDataService())
+    }
+    
+    func testFillChartDataWithNilGoals() {
+        // Given
+        viewModel.allGoals = nil
+        // When
+        let chartData = viewModel.fillChartData()
+        // Then
+        XCTAssertNil(chartData)
+    }
+    
+    func testFillChartDataWithEmptyGoals() {
+        // Given
+        viewModel.allGoals = []
+        // When
+        let chartData = viewModel.fillChartData()
+        // Then
+        XCTAssertNil(chartData)
+    }
+    
+    func testFillChartDataWithGoals() {
+        // Given
+        let mockGoals = viewModel.allGoals!
+        // When
+        let chartData = viewModel.fillChartData()
+        // Then
+        XCTAssertNotNil(chartData)
+        XCTAssertEqual(chartData?.count, mockGoals.count)
```

Comment:
I would suggest to add a bit more test such as "expected bar height"

Reply:
So basically trying to merge this test with the one of `calculateNumberOfCompletedTasks(goal:)` and the one of `calculateGoalProgress(goal:).

At the end, the test should look like this piece of code I did, with eventually even more assert:
```
// Given
let mockGoals: [Goal] = [
    Goal(
        name: "Test goal 2",
        createdAt: Calendar.current.date(byAdding: .day, value: -2, to: Date())!,
        isCompleted: false,
        tasks: [
            Task(name: "Test task 2.1", isCompleted: true),
            Task(name: "Test task 2.2", isCompleted: false),
            Task(name: "Test task 2.3", isCompleted: true)
        ]
    ),
    Goal(
        name: "Test goal 1",
        createdAt: Calendar.current.date(byAdding: .day, value: -1, to: Date())!,
        isCompleted: true,
        tasks: [
            Task(name: "Test task 1.1", isCompleted: true),
            Task(name: "Test task 1.2", isCompleted: true),
            Task(name: "Test task 1.3", isCompleted: true)
        ]
    )
]
dataService.allGoals = mockGoals // viewModel.allGoals = mockGoals in your case but I suggest you to delete ProgressViewModel.allGoals

// When
let chartData = viewModel.fillChartData()

// Then
XCTAssertEqual(chartData?.count, mockGoals.count)

XCTAssertEqual(chartData?[0].legend, Legend(color: .yellow, label: "Big Progress", order: 4))
XCTAssertEqual(chartData?[0].startValue, 0)
XCTAssertEqual(chartData?[0].endValue, 3)
XCTAssertEqual(chartData?[0].visible, true)
XCTAssertEqual(chartData?[0].label, "")

XCTAssertEqual(chartData?[1].legend, Legend(color: .green, label: "Success", order: 5))
XCTAssertEqual(chartData?[1].startValue, 0)
XCTAssertEqual(chartData?[1].endValue, 4)
XCTAssertEqual(chartData?[1].visible, true)
XCTAssertEqual(chartData?[1].label, "")
```